archive/compression/compression.go

```diff
@@ -47,11 +47,14 @@ const (
 	Zstd
 )
 
-const disablePigzEnv = "CONTAINERD_DISABLE_PIGZ"
+const (
+	disablePigzEnv  = "CONTAINERD_DISABLE_PIGZ"
```

Comment:
wondering if we should start looking at a (comma separated) list of implementations to use, and use that to enable/disable each (and set priority)

Reply:
Based on the benchmarks we have, the performance numbers are igzip > pigz > Go's compress/gzip. People enabling/disabling specific implementations should be rare, I think.

I checked GitHub to see people who have disabled pigz. Interestingly Nydus was the one, but it was enabled after :)

https://github.com/dragonflyoss/image-service/pull/1117/files

Reply:
If we break backward-compatibility, does it have to be in environment variables?

Reply:
Possible options are

- disable_external_gzip = true/false - Basically what @samuelkarp suggested
- external_gzip = ["igzip", "unpigz"] - Being extensible is good, but may be too much? It could have `[]` to say "Don't use external commands"

Reply:
If we pick this config.toml path. I'm gonna remove CONTAINERD_DISABLE_PIGZ. It is for 2.0 and the environment variable wasn't documented.

Reply:
I think this is an environment variable as at least one call path occurs in the smart client (via [import.go](https://github.com/containerd/containerd/blob/40099ed56b90fdf0bb668a381430243849828351/import.go#L138)) and we don't have `config.toml` for the client.